src/merge/merge-artifacts.ts

```diff
@@ -32,7 +33,28 @@ export async function run(): Promise<void> {
   )
 
   if (artifacts.length === 0) {
-    throw new Error(`No artifacts found matching pattern '${inputs.pattern}'`)
+    // No files were found, different use cases warrant different types of behavior if nothing is found
+    switch (inputs.ifNoFilesFound) {
```

Comment:
I would definitely add a default behavior of the switch statement emitting an error in case the value is something not yet know, even though the values are validated before. Just to make sure somebody implementing a new option can not forget it.